portfolio/src/main/webapp/script.js

```diff
@@ -53,11 +53,18 @@ function bubbleFall() {
  * @param {ArrayList<String>} comments
  */
 function getComments() {
-  fetch('/data').then(response => response.json()).then((comments) => {
+  const baseURL = '/data';
+  let params = (new URL('https://' + location.hostname + baseURL)).searchParams;
+  const userNumComments = document.getElementById('number-of-comments').value;
+  params.append('number-of-comments', userNumComments);
+
+  console.log(params.toString());
+  fetch(baseURL + '?' + params.toString()).then(response => response.json()).then((comments) => {
```

Comment:
```suggestion
  let url = new URL('https://' + location.hostname + '/data');
  const userNumComments = document.getElementById('number-of-comments').value;
  url.searchParams.append('number-of-comments', userNumComments);

  console.log(params.toString());
  fetch(url.toString()).then(response => response.json()).then((comments) => {
```

Sorry, since you went to the trouble of using URL and the params, would this suggestion work?

Reply:
Sounds good!